### PR TITLE
Error in tags on validation page.

### PR DESCRIPTION
**Pull request by nader77**
#343

### CODE REVIEW

**Comment by nader77** [comment]
Update: Removing duplicated code, Placing the code in functions.

**Comment by nader77** [comment]
Update: Fixed the app js settings.

**Comment by nader77** [comment]
Update: Added a function to show the selected tags on the edit page upon validation.
selection_049

**Comment by nader77** [comment]
@amitaibu Is this the correct way or should i do it in an another way ?

File: capacity4more/modules/c4m/og/c4m_og_vocab/c4m_og_vocab.module [hunk 1]
```diff
@@ -258,6 +240,41 @@ function c4m_og_vocab_field_widget_form(&$form, &$form_state, $field, $instance,
   return $element;
 }
 
+function c4m_og_vocab_validate($element, &$form_state, $form) {
```
Comment (amitaibu) [inline capacity4more/modules/c4m/og/c4m_og_vocab/c4m_og_vocab.module]: PHP docs.

File: capacity4more/modules/c4m/og/c4m_og_vocab/c4m_og_vocab.module [hunk 2]
```diff
@@ -258,6 +240,41 @@ function c4m_og_vocab_field_widget_form(&$form, &$form_state, $field, $instance,
...
+  $values = $form_state['values']['og_vocabulary'][LANGUAGE_NONE][0];
+
+  $tags = array();
+  foreach ($values as $value) {
+    if (is_array($value)) {
+      foreach ($value as $term_array) {
+        if ($term_array['target_id']) {
+          if($term_array['target_id'] == 'autocreate') {
+            $tags[] = array(
+              'isNew' => true,
```
Comment (amitaibu) [inline capacity4more/modules/c4m/og/c4m_og_vocab/c4m_og_vocab.module]: true => TRUE (here and below)

File: capacity4more/modules/c4m/vocab/c4m_vocab_widget/c4m_vocab_widget.module [hunk 3]
```diff
@@ -112,6 +80,71 @@ function c4m_vocab_widget_field_widget_form(&$form, &$form_state, $field, $insta
...
+    // If the term is array, This vocab is of type "node".
+    if(is_array($term)) {
+      foreach($term as $id => $label) {
+        $options[$id]['label'] = check_plain($label);
+        $options[$id]['id'] = $id;
+      }
+    }
+    else {
+      $parents = $term->parents;
+      if ($parents[0] == '0') {
```
Comment (amitaibu) [inline capacity4more/modules/c4m/vocab/c4m_vocab_widget/c4m_vocab_widget.module]: add comments

Comment (HelenaEksler) [inline capacity4more/modules/c4m/vocab/c4m_vocab_widget/c4m_vocab_widget.module]: Comment is on the next line.

**Comment by nader77** [comment]
Update: Hide unnecessary border under the tags input.
selection_050

**Comment by nader77** [comment]
@amitaibu PR is ready for code review.

**Pull request merged**